classes/submissionFile/DAO.inc.php

```diff
@@ -0,0 +1,465 @@
+<?php
+
+/**
+ * @file classes/submissionFile/DAO.inc.php
+ *
+ * Copyright (c) 2014-2021 Simon Fraser University
+ * Copyright (c) 2003-2021 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class DAO
+ * @ingroup submissionFile
+ *
+ * @see SubmissionFile
+ *
+ * @brief Operations for retrieving and modifying submission files
+ */
+
+namespace PKP\submissionFile;
+
+use APP\core\Application;
+use APP\facades\Repo;
+
+use Exception;
+use Illuminate\Support\Collection;
+use Illuminate\Support\Facades\App;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\LazyCollection;
+
+use PKP\core\EntityDAO;
+use PKP\db\DAORegistry;
+use PKP\plugins\PKPPubIdPluginDAO;
+use PKP\services\PKPSchemaService;
+use PKP\submission\reviewRound\ReviewRound;
+
+use stdClass;
+
+class DAO extends EntityDAO implements PKPPubIdPluginDAO
+{
+    /** @copydoc EntityDAO::$schema */
+    public $schema = PKPSchemaService::SCHEMA_SUBMISSION_FILE;
+
+    /** @copydoc EntityDAO::$table */
+    public $table = 'submission_files';
+
+    /** @copydoc EntityDAO::$settingsTable */
+    public $settingsTable = 'submission_file_settings';
+
+    /** @copydoc EntityDAO::$primaryKeyColumn */
+    public $primaryKeyColumn = 'submission_file_id';
+
+    /** @copydoc EntityDAO::$primaryTableColumns */
+    public $primaryTableColumns = [
+        'assocId' => 'assoc_id',
+        'assocType' => 'assoc_type',
+        'createdAt' => 'created_at',
+        'fileId' => 'file_id',
+        'fileStage' => 'file_stage',
+        'genreId' => 'genre_id',
+        'id' => 'submission_file_id',
+        'sourceSubmissionFileId' => 'source_submission_file_id',
+        'submissionId' => 'submission_id',
+        'updatedAt' => 'updated_at',
+        'uploaderUserId' => 'uploader_user_id',
+        'viewable' => 'viewable',
+    ];
+
+    /**
+     * Instantiate a new SubmissionFile
+     */
+    public function newDataObject(): SubmissionFile
+    {
+        return App::make(SubmissionFile::class);
+    }
+
+    /**
+     * @copydoc EntityDAO::get()
+     */
+    public function get(int $id): ?SubmissionFile
+    {
+        $row = DB::table($this->table . ' as sf')
+            ->join('submissions as s', 's.submission_id', '=', 'sf.submission_id')
+            ->join('files as f', 'f.file_id', '=', 'sf.file_id')
+            ->where($this->primaryKeyColumn, '=', $id)
+            ->addSelect(['sf.*', 'f.*', 's.locale as locale'])
```

Comment:
I want to revisit @asmecher [comment](https://github.com/pkp/pkp-lib/pull/7166/files/9532058ec064abe544b625c50821da55e36f41ae#diff-1ab22d4d9db06416915a2c05a7209dc7107d2d7249313a383f2d7fdcec7ffc42):

> The `Collector` already has a `select()` clause specifying `sf.*` (or it should be once that is corrected -- see other comment), so I would suggest using `addSelect(['f.*', 's.locale as locale'])` here. But since these additional joins are required for `fromRow` to work, why not move them into the `getQueryBuilder` function entirely? 

The way I did it was to include the additional required joins in the `getQueryBuilder` function. In hindsight, however, it is probably more performant not to do this. That way a call to `getIds` or `getCount` will not include unnecessary join/select statements.

To do this in a nice way, we'd have to extend the `Illuminate\Database\Query` class for every entity, as well as the `DB` facade, which seems like more abstraction than is beneficial in this case. The other workaround is a little bit ugly but works:

```php
public function selectObject(): Builder
{
    return $this
        ->join('submissions as s', 's.submission_id', '=', 'sf.submission_id')
        ->join('files as f', 'f.file_id', '=', 'sf.file_id')
        ->select(['sf.*', 'f.*', 's.locale as locale']);
}

public function get(int $id): ?SubmissionFile
{
    $q = DB::table($this->table . ' as sf')
        ->where($this->primaryKeyColumn, '=', $id);
    $row = $this->selectObject($q)
        ->first();
    return $row ? $this->fromRow($row) : null;
}
```

This would provide us a way to prevent additional DB calls in the `fromRow` method, which are pretty common and not very performant.